rust/crd/src/lib.rs

```diff
@@ -79,6 +79,7 @@ impl HiveRole {
 #[serde(rename_all = "camelCase")]
 pub struct MetaStoreConfig {
     pub warehouse_dir: Option<String>,
+    #[serde(default)]
     pub database: DatabaseConnectionSpec,
```

Comment:
Are we comfortable defaulting to Derby here? Wouldn't that cause issues as soon as you try scaling out hive?

Reply:
Agreed that we want a nice quick start experience, but IMO that's kinda covered by the example YAML?

I'm just a bit wary of hidden defaults that end up locking the user into a setup with certain tradeoffs (without getting into a non-trivial database migration, at least).

Reply:
I agree. The problem is it needs a default (because of the current operator-rs impl). For what its worth, derby is the easiest to set up. I dont see another proper default here?

Reply:
Failure is an option for invalid config :P

That said, yeah, ergonomically that's kind of problematic atm (both for users until we have Event support, and for devs w/ the current APIs).

I have started to prototype the latter a bit, but it's definitely still in an experimental stage.